vochain/process.go

```diff
@@ -63,49 +68,66 @@ func (v *State) Process(pid []byte, isQuery bool) (*models.Process, error) {
 	var processBytes []byte
 	var err error
 	v.RLock()
-	if isQuery {
-		if processBytes, err = v.Store.ImmutableTree(ProcessTree).Get(pid); err != nil {
-			return nil, err
-		}
-	} else {
-		if processBytes, err = v.Store.Tree(ProcessTree).Get(pid); err != nil {
-			return nil, err
-		}
-	}
+	processBytes, err = v.mainTreeViewer(isQuery).
+		DeepGet([]*statedb.TreeConfig{ProcessesCfg}, pid)
 	v.RUnlock()
-	if processBytes == nil {
+	if err == arbo.ErrKeyNotFound {
 		return nil, ErrProcessNotFound
+	} else if err != nil {
+		return nil, err
 	}
-	process := new(models.Process)
-	err = proto.Unmarshal(processBytes, process)
+	var process models.StateDBProcess
+	err = proto.Unmarshal(processBytes, &process)
 	if err != nil {
 		return nil, fmt.Errorf("cannot unmarshal process (%s): %w", pid, err)
 	}
-	return process, nil
+	return process.Process, nil
 }
 
 // CountProcesses returns the overall number of processes the vochain has
-func (v *State) CountProcesses(isQuery bool) int64 {
-	v.RLock()
-	defer v.RUnlock()
-	if isQuery {
-		return int64(v.Store.ImmutableTree(ProcessTree).Count())
+func (v *State) CountProcesses(isQuery bool) (uint64, error) {
+	// TODO: Once statedb.TreeView.Size() works, replace this by that.
```

Comment:
Side note: we will need to take this into account depending on the usage of this method, as it can be slow to compute with the current approach.